x-pack/plugins/observability/public/pages/slo_edit/components/common/query_builder.tsx

```diff
@@ -37,12 +38,19 @@ export function QueryBuilder({
     },
   } = useKibana().services;
 
-  const { control, getFieldState } = useFormContext<CreateSLOForm>();
+  const { control, getFieldState, watch } = useFormContext<CreateSLOForm>();
 
   const { dataView } = useCreateDataView({
     indexPatternString,
   });
 
+  const valueText = watch(name) as string;
+  const [inputVal, setInputVal] = useState<string>(valueText);
+
+  useEffect(() => {
+    setInputVal(valueText);
+  }, [valueText]);
+
```

Comment:
At first I thought this was sub optimal as we are watching the value so we can store it in a state everytime it changes. But on second read I understand where it came from.